packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -229,6 +232,10 @@ class AndroidSdkVersion implements Comparable<AndroidSdkVersion> {
   String get apksignerPath => getBuildToolsPath('apksigner');
 
   List<String> validateSdkWellFormed({bool requireApkSigner = true}) {
+    if (buildToolsVersion.major < minimumAndroidSdkVersion) {
```

Comment:
This describes how the legacy build works. We're switching to Gradle-based builds, where flutter tools don't get to select the SDK or build tools versions. Instead, Gradle uses the versions specified in the build.gradle file, so it doesn't really matter what version flutter thinks it has 'selected'.

Gradle will also automatically download the versions of build tools and SDK it needs, so there's no point in checking that they're installed before we call Gradle.

We currently still support the legacy builds, where this check (and the SDK/build tools selection) matters, but once we remove that and switch completely to Gradle (#8455), all we need to check is that the SDK/build tools licenses have been accepted (#8093).